alexa_media/__init__.py

```diff
@@ -257,14 +259,20 @@ def update_devices():
                              [email]
                              ['entities']
                              ['media_player'].values())
+        if (hass.data[DATA_ALEXAMEDIA]['accounts'][email]['websocket']
+                and not (hass.data[DATA_ALEXAMEDIA]
+                         ['accounts'][email]['new_devices'])):
+            return
+        hass.data[DATA_ALEXAMEDIA]['accounts'][email]['new_devices'] = False
         devices = AlexaAPI.get_devices(login_obj)
         bluetooth = AlexaAPI.get_bluetooth(login_obj)
         _LOGGER.debug("%s: Found %s devices, %s bluetooth",
                       hide_email(email),
                       len(devices) if devices is not None else '',
                       len(bluetooth) if bluetooth is not None else '')
         if ((devices is None or bluetooth is None)
-                and not hass.data[DOMAIN]['accounts'][email]['config']):
+                and not hass.data[DATA_ALEXAMEDIA]
+                                 ['accounts'][email]['config']):
```

Comment:
I'd prefer, but perhaps in the next version to remove the scan interval stuff and throttle to make this entirely event driven. Should clean up the code considerably when we can just fire on events.